### PR TITLE
feat: landing page for relay server

### PR DESCRIPTION
**Pull request by ZinoKader**
Just a fun landing page to show something when one hits the `/` path on the relay server with a browser. Probably something most people that start up a relay do the first time! Also, the [portal.spatiumportae.com](https://portal.spatiumportae.com) domain is going to pop up in the logs every time you send/receive something.

Currently shows the Portal logo, the relay's version, and a (pretty cool) animated "star travel" screensaver type of background.

Clicking the logo brings you to the Portal GitHub page. Clicking the version brings you to the releases page.

Hovering over the logo/version puts the animation into hyperdrive, taking off into hyperspace. It's a lot of fun!

Changes:
- Template parsing that embeds templates into the binary, so that no particular directories are needed alongside the binary.
- Template file for landing page: https://github.com/SpatiumPortae/portal/blob/f/server-landing/templates/tmpl/relay/landing.html
- Added handler for `/`, intialized templates in `server.go`

## Demo

https://user-images.githubusercontent.com/6842167/223131145-6ec08c8b-ff28-473a-81bc-9624ffb3f6b6.mp4

### CODE REVIEW

**Comment by ZinoKader** [comment]
Looking into embedding the templates into the binary.

File: cmd/portal/commands/receive.go [hunk 1]
```diff
@@ -150,7 +150,7 @@ func handleReceiveCommandRaw(version string, password string) error {
 		case errors.Is(err, io.EOF):
 			return nil
 		case errors.Is(err, file.ErrUnpackFileExists):
-			fmt.Printf("overwrite %s? [y/n] ", committer.FileName())
+			fmt.Printf("overwrite %s? [Y/n] ", committer.FileName())
```
Comment (mellonnen) [inline cmd/portal/commands/receive.go:153]: why not capitalize _n_ if you capitalize _y_?

**Review by mellonnen — CHANGES_REQUESTED** [review]
🚀 🌃

File: .gitignore [hunk 2]
```diff
@@ -12,6 +12,9 @@
 # Portal log files
 .portal-*.log
 
+# Portal built binary
+portal
+
```
[the remark this answers is not in the document]
Comment (ZinoKader) [reply .gitignore:17]: Ah thanks!

[the remark this answers is not in the document]
Comment (mellonnen) [reply .gitignore:17]: Should also unignore Dockerfile!

[the remark this answers is not in the document]
Comment (ZinoKader) [reply .gitignore:17]: This does not ignore the (by default) built binary called portal, though?

File: cmd/portal/commands/receive.go (hunk 1 above)
Comment (ZinoKader) [reply cmd/portal/commands/receive.go:153]: This is the de-facto standard of showing the default selection in cli programs, as I have seen. It's the same in the TUI, although there, you can also use the left/right arrow keys to change the selection, which is shown by which letter is capitalized.

File: Dockerfile [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 # Copy binary from build container and build image.
 FROM alpine:3.14
-RUN mkdir /usr/app 
+RUN mkdir /usr/app
 WORKDIR /usr/app
-COPY --from=build-stage /usr/app/app .
+COPY --from=build-stage /usr/app/portal .
 
-ENTRYPOINT [ "./app", "serve","-p", "8080" ]
+ENTRYPOINT [ "./portal", "serve", "-p", "8080" ]
```
[the remark this answers is not in the document]
Comment (ZinoKader) [reply Dockerfile]: ✅

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (mellonnen) [reply .gitignore:17]: It ignores all files with no extension instead of explicitly ignore the binary

[the remark this answers is not in the document]
Comment (mellonnen) [reply .gitignore:17]: That is weird 🤔

**Pull request merged**